app/controllers/CyaUpdateRecordController.scala

```diff
@@ -128,8 +129,14 @@ class CyaUpdateRecordController @Inject() (
       goodsRecordConnector
         .getRecord(request.eori, recordId)
         .flatMap { recordResponse =>
+          val navigatingFromExpiredCommCode = request.session.get(fromExpiredCommodityCodePage).contains("true")
```

Comment:
Is this necessarily needed? 

You have the goods record already so you can tell if its expired from that, rather than the page you've been :)

Reply:
yeah, I had that code earlier but due to more duplications, I wanted to keep it simple like this with a flag.